pkg/sql/executor.go

```diff
@@ -516,10 +528,22 @@ func (e *Executor) Prepare(
 	return prepared, nil
 }
 
+func (e *Executor) ExecuteStatementsBuffered(
+	session *Session, stmts string, pinfo *parser.PlaceholderInfo,
+) StatementResults {
+	// TODO:(tohlson): make a buffered writer implementation
```

Comment:
I'm unsure whether I need a new implementation or whether once I add a buffer to the streaming implementation that this problem will go away easily.